testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSyncPartitionMetadata.java

```diff
@@ -64,6 +64,33 @@ public void testAddPartition()
         cleanup(tableName);
     }
 
+    @Test(groups = {HIVE_PARTITIONING, SMOKE, TRINO_JDBC})
+    @Flaky(issue = ERROR_COMMITTING_WRITE_TO_HIVE_ISSUE, match = ERROR_COMMITTING_WRITE_TO_HIVE_MATCH)
+    public void testAddPartitionContainingCharactersThatNeedUrlEncoding()
```

Comment:
There are already tests for adding and dropping partitions, maybe it's enough to add new dirs in the `prepare()` method instead of adding more tests?